sycl/test/check_device_code/device_asan_no_ctor.cpp

```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux
+// RUN: %clangxx -fsycl-device-only -fno-sycl-early-optimizations -Xarch_device -fsanitize=address -emit-llvm %s -S -o %t.ll
+// RUN: FileCheck %s --input-file %t.ll
+
+// CHECK-NOT: asan.module_ctor
```

Comment:
Please add a comment explaining why we want that.

Reply:
done